src/Symfony/Component/Uid/UuidV1.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uid;
+
+/**
+ * @experimental in 5.1
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class UuidV1 extends Uuid
+{
+    protected const TYPE = UUID_TYPE_TIME;
+
+    // https://tools.ietf.org/html/rfc4122#section-4.1.4
+    // 0x01b21dd213814000 is the number of 100-ns intervals between the
+    // UUID epoch 1582-10-15 00:00:00 and the Unix epoch 1970-01-01 00:00:00.
+    private const TIME_OFFSET_INT = 0x01b21dd213814000;
+    private const TIME_OFFSET_COM = "\xfe\x4d\xe2\x2d\xec\x7e\xc0\x00";
+
+    public function __construct(string $uuid = null)
+    {
+        if (null === $uuid) {
+            $this->uuid = uuid_create(static::TYPE);
```

Comment:
This case is already handled in the parent constructor. Looks like the constructor can be removed from UuidV1?

Reply:
And UuidV4

Reply:
Good catch thanks, I forgot this in the parent constructor from a previous iteration.
(the parent constructor doesn't accept `null` - only v1 and v4 can be generated with not args.)